packages/lodestar/src/chain/archiver/index.ts

```diff
@@ -69,18 +69,54 @@ export class Archiver {
       await archiveBlocks(this.db, this.chain.forkChoice, this.chain.lightClientServer, this.logger, finalized);
 
       // should be after ArchiveBlocksTask to handle restart cleanly
-      await this.statesArchiver.maybeArchiveState(finalized, this.chain.anchorSlot);
-
-      await Promise.all([
-        this.chain.checkpointStateCache.pruneFinalized(finalizedEpoch),
-        this.chain.stateCache.deleteAllBeforeEpoch(finalizedEpoch),
-      ]);
+      await this.statesArchiver.maybeArchiveState(finalized);
 
+      this.chain.checkpointStateCache.pruneFinalized(finalizedEpoch);
+      this.chain.stateCache.deleteAllBeforeEpoch(finalizedEpoch);
       // tasks rely on extended fork choice
       this.chain.forkChoice.prune(finalized.rootHex);
+      void this.updateBackfillRange(finalized);
```

Comment:
While the errors are handled above, why not await for completeness? updateBackfillRange() should take too long